### PR TITLE
Feature/add link command

### PR DESCRIPTION
**Pull request by salco**
I separated the Link command to let us use it directly, it's useful when you have to switch between different build folder.

### CODE REVIEW

**Comment by ilyachur** [comment]
Hi @salco 

Thank you for contribution!

Could you please explain your scenario? How do you switch between different build folders?
Do you change manually `g:cmake_build_dir` or `g:cmake_build_type`?

I am asking it because, maybe will be better to add a command to switch between different build modes.

**Comment by salco** [comment]
I use you tool to design embedded soft, I use 3 different `CMAKE_BUILD_TYPE`: UTest,Debug,Release.
my `g:cmake_build_dir` doesnt change it's: `"build/" . g:cmake_build_type`.
I only need to change `g:cmake_build_type` between the three type, since for me it's `CMAKE_BUILD_TYPE` that I need to change.

For example UTest is unit test build on PC and Debug &Release are for the embedded with a different gcc.
I use the link with `compile_commands.json` to the root of the project to be able to work with YouCompleteMe.
Each build type may have differant option in the `compile_commands.json`.

What do you mean by different build modes, i'm not sure it's the `CMAKE_BUILD_TYPE`?

**Comment by ilyachur** [comment]
@salco Yes, I mean maybe it is better to add a command to switch between different build types. Like:
```
:CMakeSelectBuildType
or
:CMakeSwitchBuildType
or
:CMakeChangeBuildType
or
:OtherVariant
```

And this command should change the build directory, link the `compile_commands.json` and etc. From my point of view it will be better than command which only links a `compile_commands.json`.

But in this case need to think how to change the build directory in your case, because you use the custom build path (by default build filder is `cmake-build-<cmake_build_type>`) which uses global variables.

PS:

Also may I ask you to also update [this document](https://github.com/ilyachur/cmake4vim/blob/master/doc/cmake4vim.txt) if you add a new command?

**Comment by salco** [comment]
@ilyachur okok, I will update the doc when we get a better idea of the workflow. 

- So if i call i.e. `CMakeSelectBuildType Debug`, the function will set `g:cmake_build_type` 
- After that if I call `CMake `  it will build like before and if the variable `g:cmake_compile_commands_link` is correct it will do the link.

Personally I don't use the default build folder to avoid getting 9 new folder on the root of my project.
I think if your ok with the example above it should work for both build setup.

**Comment by ilyachur** [comment]
As I see, you locally define the build directory.
If `CMakeSelectBuildType Debug` change the `g:cmake_build_type`, it is not solve the issue that you also manually should update the build folder before the call `:CMake`

From my point of view if user uses the default path for cmake build directory the command `:CMakeSelectBuildType` also should update the build folder and after that do the link if `compile_commands.json` exists.

**Comment by salco** [comment]
the command `:CMake` call `cmake4vim#GenerateCMake` that create the build folder if I understand correctly?

So `:CMakeSelectBuildType` should do:
* change `g:cmake_build_type`
* call `:CMake`, (it will create the folder and do the command if its needed)
* link the `compile_commands.json` if it exist and `g:cmake_compile_commands_link` is correct

Is the proposition correct, or we need to change some point?

Also, I've just re-check the code and maybe just removing `g:cmake_compile_commands &&`, here: 
https://github.com/ilyachur/cmake4vim/blob/master/autoload/cmake4vim.vim#L122

Could fix my main issue, for me the `CMAKE_EXPORT_COMPILE_COMMANDS=ON` is in my CMakeFiles.txt. 
I think it's an extra check that is not needed at that point, if there is a `compile_commands.json` and  `g:cmake_compile_commands_link` is correct `cmake4vim` should link the file.

@ilyachur Tell me if you prefer the command or the change at https://github.com/ilyachur/cmake4vim/blob/master/autoload/cmake4vim.vim#L122.

**Comment by ilyachur** [comment]
@salco Yes you are right that `cmake4vim#GenerateCMake` can create the build folder.

I agree with your summary about `:CMakeSelectBuildType` command, from my point of view it looks like a complete solution, do you agree?

About the line [122](https://github.com/ilyachur/cmake4vim/blob/master/autoload/cmake4vim.vim#L122 and this conditional, I agree that the `g:cmake_compile_commands` check is redundant. And it is a valid change.

Also I already thought before that it is good to have a command which will be automatically switch build configurations. And it will be good if we add this command.

I like your idea about `g:cmake_build_dir_prefix . g:cmake_build_type`. In this case the `:CMakeSelectBuildType` command should work for all scenarios: by default `g:cmake_build_dir_prefix` is `cmake-build-`, for your case `g:cmake_build_dir_prefix` is a `build/`.

But also for backward compatibility we should have a `g:cmake_build_dir` variable and if user defines this variable use it. Something like:
```
let l:cmake_build_dir = g:cmake_build_dir_prefix . g:cmake_build_type
if g:cmake_build_dir
     let l:cmake_build_dir = g:cmake_build_dir
```

**Comment by salco** [comment]
@ilyachur  I've updated the code with all we have talked about. 
The markdown and help file is updated with the new info.

File: test/3_cmake_targets.vader [hunk 1]
```diff
@@ -9,7 +9,7 @@ Before:
 
     function! RemoveCMakeDirs()
         if has('win32')
-            for l:value in ['cmake-build-release', 'cmake-build-debug', 'cmake-build-relwithdebinfo', 'cmake-build-minsizerel', 'cmake-build', 'build_test']
+            for l:value in ['cmake-build-Release', 'cmake-build-Debug', 'cmake-build-relwithdebinfo', 'cmake-build-minsizerel', 'cmake-build', 'build_test']
```
Comment (ilyachur) [inline test/3_cmake_targets.vader:12]: You have done backward incompatible changes when you removed `toLower()` call for build type.

In this line you also need to fix `relwithdebinfo` and `minsizerel` in this case.

**Review by ilyachur — APPROVED** [review]

**Comment by ilyachur** [comment]
@salco Thank you for contributing!

I have merged this PR to master.

**Pull request merged**